### ISSUE TITLE
Status on Project

### ISSUE THREAD

**Issue by BobtheBob123456** [bundabrg/GeyserReversion#64]
Hey Guys,
I just wanted to know what the status of this project is. Is it close to completion, or is it going to be a while before it is released? Do you have an approximate date of when this will be released?

Also, will the plugin work where you will run a command to generate a code, which the education edition player will use to join the server, or will it work in a different way.

Thanks,
Ishaan

Also, it would be nice if you added education edition resource pack support as well.

**Comment by bundabrg** [comment]
Its taking a bit as I'm having to allocate attention to multiple things at the moment but I am still working on it. I can't give an ETA yet.

Regarding the code, yes it will generate a code (as that appears to be the only way to connect in new versions) but unfortunately the code will change over time as it expires, but the server will either auto-renew a new one or if the API permits it, renew the old code (I'm not sure if this is possible or not). So we'll need to then have a way to extract the code to something like a webpage so maybe it writes it to a file or triggers an event that then the server owner can use to update a webpage. As I'm adding plugin support to it a plugin may be the best option that does the required action.

**Comment by Windswipe** [comment]
@bundabrg Here's an idea: Maybe make the code automatically sent in the form of a Discord webhook or have a bot that updates the code? Neither would require too much work on your end. (At least from my basic assumptions)

**Comment by p0t4t0sandwich** [comment]
Just following up, as I have no clue where to start with generating the join code. (got the token stuff working in a basic extension plugin however, and I'll be testing with a version of Geyser that might roughly support the bedrock version the current edu release is based on)
To my understanding the user broadcasts themselves to Microsoft's discovery system, which then generates the join code, and then other clients lookup what IP:port the join code is associated with.

**Comment by josef240** [comment]
I could tell you how to generate a join code.

**Comment by p0t4t0sandwich** [comment]
Just to clarify, I'm talking about the 4-symbol join code (and associated longer integer ID) that the client uses to query MS's services for the "server's" IP and port info. I'm all ears if you know of any resources/documentation that go over the join-code generation. From my research I think it may use playfab as part of the process.

**Comment by josef240** [comment]
Yes, it uses Playfab and a separate API. Using a VALID Microsoft token, you can add a Nethernet world to DIscovery. If someone queries for it, it would pop up in ONLY your tenant. IDK how you would do it for a Raknet server though.

**Comment by p0t4t0sandwich** [comment]
Mind linking to some documentation so I can read up on the topic?

**Comment by josef240** [comment]
I'm making documentation, I didn't find this info online.

**Comment by josef240** [comment]
Its released
